src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -243,6 +244,24 @@ export const useDiscoverHistogram = ({
     columns: savedSearchData$.documents$.getValue().textBasedQueryColumns ?? [],
   });
 
+  useEffect(() => {
+    if (!isPlainRecord) {
+      return;
+    }
+
+    const fetchStart = stateContainer.dataState.fetch$.subscribe(() => {
+      setIsSuggestionLoading(true);
```

Comment:
```suggestion
      if (!skipRefetch.current) {
        setIsSuggestionLoading(true);
      }
```
It looks like we still have the double loader issue with this solution, but it looks like checking for `skipRefetch` which is `true` when the chart is unhidden fixes it.

Reply:
Done https://github.com/elastic/kibana/pull/165830/commits/84e6bc07701a6864a6f29bef50842af5c270626e

I could not reproduce it when I was refactoring it but I cant reproduce it either with this change

Reply:
I was only able to do it while using request throttling, but it's fixed now, thanks!